cgrep.py

```diff
@@ -0,0 +1,443 @@
+# Copyright 2011 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# Simple util to grep through network and service definitions.
+# Examples:
+#   To find out which tokens contain "10.4.3.1" use
+#   $ cgrep.py -i 10.4.3.1
+#
+#   To find out if token 'FOO' includes ip "1.2.3.4" use
+#   $ cgrep.py -t FOO -i 1.2.3.4
+#
+#   To find the difference and union of tokens 'FOO' and 'BAR' use
+#   $ cgrep.py -c FOO BAR
+#
+#   To find the difference of network tokens to which 2 IPs belong use
+#   $ cgrep.py -g 1.1.1.1 2.2.2.2
+#
+#   To find which IPs are in the 'FOO' network token use
+#   $ cgrep.py -o FOO
+#
+#   To find which port & protocol pairs are in a service token 'FOO' use
+#   $ cgrep.py -s FOO
+#
+#   To find which service tokens contain port '22' and protocol 'tcp' use
+#   $ cgrep.py -p 22 tcp
+#
+
+import argparse
+import pprint
+from lib import naming
+from lib import nacaddr
+
+
+__author__ = ("watson@google.com (Tony Watson)", "jhrbek@fb.com (Joe Hrbek)"
```

Comment:
Do you really want to be added as an author here? I believe authors are no longer required per Googles code standards and I have been contemplating removing all __author__ tags everywhere. Git maintains a much better history and can show exactly how much someone has contributed. I am not saying you do not deserve credit on this file. You have changed a very significant portion of this file. I am wondering what your opinions are on authors tags in general since this is the first time I have seen someone add themselves to the authors.